docs/howto/write.rst

```diff
@@ -1,65 +1,121 @@
-- Write with secret prompt
+Secrets in the vault are either written manually or generated by secret engines (see
+:ref:`secret-engines`). ``vault-cli`` provides primitives for writing secrets manually.
 
-Write a secret
---------------
+.. code:: console
+
+   $ vault set a b=c
+   Done
+   $ vault get a
+   ---
+   b: c
+
+Managing several secrets in a secret object
+-------------------------------------------
+
+You can write several keys at once:
 
 .. code:: console
 
-   $ vault set my_other_secret value=supersecret
+   $ vault set a b=c d=e
    Done
+   $ vault get a
+   ---
+   b: c
+   d: e
 
-Write a secret via ``stdin``
-----------------------------
+By default, specified keys are added to the existing secret objects. If you want
+to redefine a secret object from scratch, use ``--clear``:
 
-You can use this when the secret has multiple lines or starts with a ``-``:
+.. code:: console
+
+   $ vault set a f=g
+   Done
+   $ vault get a
+   ---
+   f: g
+
+Write a secret from files and ``stdin``
+---------------------------------------
+
+If you assign the special value ``-`` to a secret, its value will be read from
+``stdin``. You can use this when the secret has multiple lines or starts with a ``-``:
 
 .. code:: console
 
-   $ vault set third_secret certificate=-
+   $ vault set http certificate=-
    ----BEGIN SECRET KEY----
    ...
    <hit ctrl+d to end stdin>
    Done
 
-   vault get third_secret
+   vault get http certificate
    ----BEGIN SECRET KEY----
    ...
 
-Identically, piping allows you to write the content of a file into the
-vault:
+Identically, stream redirection and pipes allows you to write the content of a file or
+the output of a program into the vault:
 
 .. code:: console
 
-   $ cat my_certificate.key | vault set third_secret certificate=-
+   $ vault set http certificate=- < my_certificate.key
    Done
 
-You can also load a key/value mapping in YAML or JSON format from a
+   $ # Or:
+   $ pwgen 64 1 | vault set myapp secret_key=-
+   Done
+
+You can also load a several key/values in YAML (or JSON) format from a
 file:
 
 .. code:: console
 
-   $ vault set third_secret --file=secret.yaml
+   $ vault set whole_secret --file=secret.yaml
    Done
 
-The special value ``--file=-`` means that the file is read from
+Identically, the special value ``--file=-`` means that the file is read from
 ``stdin``.
 
 
 Write a secret using an invisible input prompt
 ----------------------------------------------
 
 This will avoid your secrets to be displayed in plain text in your shell
-history.
+history nor on your screen.
+
+.. note::
+
+   This would only be useful when launched by a human, the ``stdin`` capabilities
+   described above are more useful for machines.
 
 .. code:: console
 
    $ vault set mypath --prompt mykey
    Please enter a value for key `mykey` of `mypath`:
    Done
+
+Caveats
+-------
+
+Safe write
+~~~~~~~~~~
+Behaviour around overwriting secrets can be controlled, see :ref:`safe-write`.
```

Comment:
Isn't this a ref to itself?

Reply:
No, it points to https://vault-cli.readthedocs.io/en/latest/howto/safe_write.html